boards: stm32n6: Add serial boot variant (#84869)
zephyrproject-rtos/zephyr: boards/st/nucleo_n657x0_q/doc/index.rst

         Build and flash an application using ``nucleo_n657x0_q`` target.

         .. zephyr-app-commands::
            :zephyr-app: samples/hello_world
            :board: nucleo_n657x0_q
            :goals: build flash

.. note::
            For flashing, before powering the board, set the boot pins in the following configuration:

            * BOOT0: 0
            * BOOT1: 1

            After flashing, to run the application, set the boot pins in the following configuration:

            * BOOT1: 0

	    Power off and on the board again.

         Run a serial host program to connect with your Nucleo board:

.. code-block:: console

   $ minicom -D /dev/ttyACM0

      .. group-tab:: Serial Boot Loader (USB)

         Additionally, connect the NUCLEO-N657X0-Q to your host computer using the USB port.
         In this configuration, ST-Link is used to power the board and for serial communication
         over the Virtual COM Port.

         .. note::
            Before powering the board, set the boot pins in the following configuration:

            * BOOT0: 1
            * BOOT1: 0

         Build and load an application using ``nucleo_n657x0_q/stm32n657xx/sb`` target (you

>          can also use the shortenned form: ``nucleo_n657x0_q//sb``)

Reviewer: ```suggestion
         can also use the shortened form: ``nucleo_n657x0_q//sb``)
```

Author: Fixed, thanks

.. zephyr-app-commands::
   :zephyr-app: samples/hello_world
   :board: nucleo_n657x0_q
   :goals: build flash

> Run a serial host program to connect with your Disco board:

Reviewer: ```suggestion
Run a serial host program to connect to you board:
```

Author: ok, fixed

Programming and Debugging
*************************

> STM32N6570_DK board includes an ST-LINK/V3 embedded debug tool interface.

Reviewer: ok so what's the name of that board then? :) 
Overview says "Nucleo-144", the board's `full_name` is Nucleo N657x0-Q, and here you have something different? (and then ST website seems to have NUCLEO-N657X0-Q with capital X btw :) )

Author: Copy pasta. Reverting back to NUCLEO_N657X0_Q

Reviewer: good :) please fix `full_name` too to use capital X, if you have a chance
